### ISSUE TITLE
source:opening_hours

### ISSUE THREAD

**Issue by matkoniecz** [mnalis/StreetComplete-taginfo-categorize#30]
https://github.com/streetcomplete/StreetComplete/issues/6057

**Comment by matkoniecz** [comment]
I found

https://github.com/mnalis/StreetComplete-taginfo-categorize/blob/master/keys.txt#L615

that sadly links to huuuuuuuuge discussion. What was the summary of it and why source keys are kept?

Though in general, when any key is removed, say XABC, then surely also source:XABC should be removed?

What if source:opening_hours is present but not opening_hours? Should it be removed then?

@jdejaegh - maybe you know answers to some of this questions - if not then you will know there is some attempt at making progress

**Comment by jdejaegh** [comment]
I just finished reading the discussion at: https://github.com/streetcomplete/StreetComplete/issues/3244

From what I got, this repository (mnalis/StreetComplete-taginfo-categorize) is there to avoid removing tags we don't understand from places when they change.  This prevents collateral damage created by StreetComplete when mapping one aspect (the shop has changed, for example).

From the discussion, it makes sense that keys starting with `source:*` are generally kept. 

In the same idea as the discussion, I think removing `source:opening_hours` when removing `opening_hours` is reasonable.  We know what the tag is used for (it's documented), and there is a rationale to remove it (it no longer applies if `opening_hours` is gone).

**Comment by mnalis** [comment]
> Though in general, when any key is removed, say XABC, then surely also source:XABC should be removed?

Yes. It is actually relatively easy to do programmatically (i.e. for every `XABC` tag to be removed, I could also generate `source:XABC` and add it to a list of tags to be removed[^1]); but the net result would be that the **list to be imported to SC would _double_ in size**, making it significantly more untidy, so I'm not sure how happy @westnordost would be with it.

I think it would be easier / cleaner if the StreetComplete code which goes through list of tags to remove, for each tag it removes not only `${tag}` but also `source:${tag}` (e.g. not only `XABC` but also `source:XABC`). It should be relatively simple change to StreetComplete; if @westnordost agrees? 

> What if source:opening_hours is present but not opening_hours? Should it be removed then?

It sounds like a broken data, but nevertheless, I think it should be removed -- when shop is gone, anything related to its previously-used opening_hours should be gone too (even if say in its last months of life someone removed `opening_hours` as they were unknown, but forgot to remove `source:opening_hours`).

[^1]: it could even be optimized in size somewhat (at a cost of even more readability), i.e. by using regex `(source:)?XABC` instead of `XABC` & `source:XABC`.

**Comment by matkoniecz** [comment]
> but nevertheless, I think it should be removed 

Then it needs to go on list, or require even more special logic in SC

And I am pretty sure that `@westnordost` will prefer longer list over extra custom logic.

**Comment by westnordost** [comment]
no, actually, i agree with mnalis

**Comment by kmpoppe** [comment]
@mnalis please update
https://github.com/mnalis/StreetComplete-taginfo-categorize/blob/fdc3a7a29285a585b82cb09346e443a978a0407f/keys.txt#L166
and
https://github.com/mnalis/StreetComplete-taginfo-categorize/blob/fdc3a7a29285a585b82cb09346e443a978a0407f/keys.txt#L178
(maybe comment them out) and update streetcomplete/StreetComplete#6066, because those lines are superfluous (and also make for a weird `check_date:check_date:...` entry in the `REMOVABLE_KEYS` val :-)
